### PR TITLE
EDGE-338 Enable auto-complete, go to definition and find usages for b…

### PR DESCRIPTION
**Pull request by ionyshch-prismsl**
…oost::movelib::unique_ptr type variables in QtCreator

### CODE REVIEW

File: src/client.cpp [hunk 1]
```diff
@@ -194,22 +194,25 @@ bool hasIntMember(const rapidjson::Value& value, const char* name)
     return value.HasMember(name)  &&  value[name].IsInt();
 }
 
+#define CREATE_SESSION() \
+    CurlSessionPtr sessionPtr = createSession();\
+    if (!sessionPtr)\
+        return makeError();\
+    CurlSession& session = *sessionPtr;
```
Comment (kyrylo-shpytsya-prism) [inline src/client.cpp]: Please see [here](http://stackoverflow.com/questions/154136/why-use-apparently-meaningless-do-while-and-if-else-statements-in-c-c-macros).

Comment (ionyshch-prismsl) [reply src/client.cpp]: Doesn't apply here, as we declare and define variables in a macro.

Comment (ionyshch-prismsl) [reply src/client.cpp]: Suggestions on improving this - doing it without macro and without code duplication or using better macro - are welcome

Comment (dbus-prsm) [reply src/client.cpp]: I don't like the idea that you have to know what `CREATE_SESSION` is doing in order to use it in the right way.
I would leave it as it was. There is no much code duplication here and that code was clear.
BTW you still have direct `createSession()` calls in the file.

Comment (Yurmusp) [reply src/client.cpp]: I do not like that variable that is used further is not transparent to the user. I think it can be re-done so that variable declaration is outside of the macro. Internally it can have another hidden variable that is "returned" by the macro if it is valid. I would not say what we had before was that bad. a bit more code, but nothing too wordy or complicated.

**Review by Yurmusp — DISMISSED** [review]

**Review by bvnp44 — APPROVED** [review]
